spring-cloud-kubernetes-loadbalancer/src/main/java/org/springframework/cloud/kubernetes/loadbalancer/KubernetesClientConfiguration.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * Copyright 2013-2020 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.loadbalancer;
+
+import io.fabric8.kubernetes.client.KubernetesClient;
+
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Bean;
+import org.springframework.core.env.Environment;
+
+/**
+ * Kubernetes load balancer client properties.
+ *
+ * @author Piotr Minkowski
+ */
+public class KubernetesClientConfiguration {
+
+	@Bean
+	@ConditionalOnProperty(name = "spring.cloud.kubernetes.loadbalancer.mode",
+			havingValue = "SERVICE")
+	KubernetesServicesListSupplier kubernetesServicesListSupplier(Environment environment,
```

Comment:
What happens in the default case, ie `POD`?

Reply:
It won't use this ListSupplier and is based on DiscoveryClient implementation directly.

Reply:
Why not have a list supplier that would get the pods?  Again we want to maintain functionality between the current ribbon implementation and the new load balancer.